src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentProperty.java

```diff
@@ -154,51 +155,86 @@ private void initDateConverter() {
 								property, actualType.getSimpleName()));
 				return;
 			}
 
-			ElasticsearchDateConverter converter = null;
+			List<ElasticsearchDateConverter> converters = new ArrayList<>();
 
-			if (dateFormat == DateFormat.custom) {
-				String pattern = field.pattern();
+			for (int i = 0; i < dateFormats.length; i++) {
+				DateFormat dateFormat = dateFormats[i];
 
-				if (!StringUtils.hasLength(pattern)) {
-					throw new MappingException(
-							String.format("Property %s is annotated with FieldType.%s and a custom format but has no pattern defined",
-									property, field.type().name()));
-				}
+				if (dateFormat == DateFormat.custom) {
+					String[] dateFormatPatterns = field.pattern();
+
+					if (dateFormatPatterns.length == 0) {
+						throw new MappingException(String.format(
+								"Property %s is annotated with FieldType.%s and a custom format but has no pattern defined.", property,
+								field.type().name()));
+					}
 
-				converter = ElasticsearchDateConverter.of(pattern);
-			} else {
-
-				switch (dateFormat) {
-					case weekyear:
-					case weekyear_week:
-					case weekyear_week_day:
-						LOGGER.warn("no Converter available for " + actualType.getName() + " and date format " + dateFormat.name()
-								+ ". Use a custom converter instead");
-						break;
-					default:
-						converter = ElasticsearchDateConverter.of(dateFormat);
-						break;
+					for (int j = 0; j < dateFormatPatterns.length; j++) {
+						String pattern = dateFormatPatterns[j];
+						if (!StringUtils.hasText(pattern)) {
+							throw new MappingException(
+									String.format("Date pattern of property %s must not be empty", property, field.type().name()));
+						}
+						converters.add(ElasticsearchDateConverter.of(pattern));
+					}
+				} else {
+
+					switch (dateFormat) {
+						case weekyear:
+						case weekyear_week:
+						case weekyear_week_day:
+							LOGGER.warn("No default converter available for " + actualType.getName() + " and date format "
+									+ dateFormat.name() + ". Use a custom converter instead.");
+							break;
+						default:
+							converters.add(ElasticsearchDateConverter.of(dateFormat));
+							break;
+					}
 				}
+
 			}
 
-			if (converter != null) {
-				ElasticsearchDateConverter finalConverter = converter;
+			if (!converters.isEmpty()) {
 				propertyConverter = new ElasticsearchPersistentPropertyConverter() {
-					final ElasticsearchDateConverter dateConverter = finalConverter;
+					final List<ElasticsearchDateConverter> dateConverters = converters;
+
+					@SuppressWarnings("unchecked")
+					@Override
+					public Object read(String s) {
+						if (isTemporalAccessor) {
+							for (ElasticsearchDateConverter dateConverter : dateConverters) {
+								try {
+									return dateConverter.parse(s, (Class<? extends TemporalAccessor>) actualType);
+								} catch (Exception e) {
+								}
+							}
```

Comment:
after the loop, if every registered converter fails, we should throw an error that none of the defined date formats could be used to read the value